sklego/meta/estimator_transformer.py

```diff
@@ -17,15 +16,18 @@ class EstimatorTransformer(TransformerMixin, MetaEstimatorMixin, BaseEstimator):
 
     :param estimator: An instance of the estimator that should be used for the transformation
     :param predict_func: The function called on the estimator when transforming e.g. (`predict`, `predict_proba`)
+    :param check_input: Whether to check the input data for NaNs, Infs and non-numeric values
     """
 
-    def __init__(self, estimator, predict_func="predict"):
+    def __init__(self, estimator, predict_func="predict", check_input=True):
         self.estimator = estimator
         self.predict_func = predict_func
+        self.check_input = check_input
 
     def fit(self, X, y, **kwargs):
         """Fits the estimator"""
-        X, y = check_X_y(X, y, estimator=self, dtype=FLOAT_DTYPES, multi_output=True)
+        X, y = check_X_y(X, y, estimator=self, force_all_finite=self.check_input, dtype=None, multi_output=True)
```

Comment:
Considering how #580 arise from #573, I personally would lean towards skipping the check entirely (and leaving it to the estimator itself) if `check_input=True` or leave it as it is otherwise. Similar to #580 the default behaviour could be `check_input=False` and delegate the check.

Here a similar example of #573 that would break if using `GroupedPredictor`:

```py
import pandas as pd

from sklearn.dummy import DummyRegressor
from sklego.meta import EstimatorTransformer, GroupedPredictor
from sklego.datasets import make_simpleseries

yt = make_simpleseries(seed=1)
dates = pd.date_range("2000-01-01", periods=len(yt))
df = (
    pd.DataFrame({"yt": yt,"date": dates})
    .assign(m=lambda d: d.date.dt.month)
    .reset_index()
)

# This works fine
_ = GroupedPredictor(DummyRegressor(), groups=["m"]).fit(df[['m']], df['yt']).predict(df[['m']])

# The following would break because GroupedPredictor requires a dataframe while the check would convert it to an array
_ = (EstimatorTransformer(GroupedPredictor(DummyRegressor(), groups=["m"]))
    .fit(df[['m', 'index']], df['yt'])
)
```
I understand that skipping _just_ a specific check is useful to your use case, but delegating it entirely feels like a more standardized solution for meta models

Reply:
I generally agree with your point about skipping checking entirely for meta models, like what was done with the Thresholder in #507. In this case why do we even bother having a `check_input` param for `GroupedPredictor` / `EstimatorTransformer`?

Among other things, many times I wanted to pass a Pandas DataFrame directly into the underlying estimator (e.g. lightgbm detects categorical variables automatically if X is a pandas dataframe) but `check_array` or `check_X_y` would convert them into vanilla numpy arrays against my wish :/

Reply:
@koaning should we conform all the _meta_ models to skip input check entirely, or at least all of them having a boolean flag to skip it? I wouldn't be against delegating everything to the model/estimator itself

Reply:
> should we conform all the meta models to skip input check entirely, or at least all of them having a boolean flag to skip it?

I'm leaning towards adding a flag to all of them. The thing is that _in general_ it is also a good idea that this check is in there. It's just that the meta models pull some advanced/unconventional stuff which is why folks hit their head. I'm not 100% sure if the default should be true or false. Both have their merit. 

The honest story here is that we probably added the `check_X_y` in all these meta models to get some of the base sklearn checks to pass in CI. Not because a user made a request.

Reply:
Sounds good, I've updated the PR accordingly. Hopefully this is good enough for merging. Thanks @FBruzzesi @koaning